packages/react-lowcode/src/codegen/generation/generators/detail/mui-detail-generator.ts

```diff
@@ -48,6 +51,155 @@ export default class MuiDetailGenerator implements DetailGenerator {
+    });
+
+    if (element) {
+      if (ts.isJsxOpeningElement(element) || ts.isJsxSelfClosingElement(element)) {
+        element.attributes.properties.forEach(prop => {
+          if (ts.isJsxAttribute(prop)) {
+            const propName = prop.name.escapedText.toString();
+
+            if (prop.initializer) {
+              if (ts.isStringLiteral(prop.initializer)) {
+                result.properties.push({
+                  name: propName,
+                  value: prop.initializer.text
+                });
+              }
+              else if (ts.isJsxExpression(prop.initializer)) {
+                if (prop.initializer.expression) {
+                  if (ts.isNumericLiteral(prop.initializer.expression)
+                    || prop.initializer.expression.kind === SyntaxKind.TrueKeyword
+                    || prop.initializer.expression.kind === SyntaxKind.FalseKeyword) {
+                    result.properties.push({
+                      name: propName,
+                      value: prop.initializer.expression.getText()
+                    });
+                  }
+                }
+              }
+            } else {
+              result.properties.push({
+                name: propName,
+                value: 'true'
+              });
+            }
+          }
+        });
+      }
+    }
+
+    return result;
+  }
+
+  async setFormWidgetProperties(position: SourceLineCol, widgetProperties: WidgetProperties): Promise<string | undefined> {
+    let result: string | undefined;
+
+    if (this._widgetContext) {
+      const sourceCode = await this._widgetContext.getSourceCodeString(position);
+      let ast = createAst(sourceCode);
+
+      if (ast) {
+        const pos = ast.getPositionOfLineAndCharacter(position.lineNumber - 1, position.columnNumber - 1);
+        const element = findByCondition<Node>(ast, (node: Node) => {
+          return node.pos === pos;
+        });
+
+        if (element) {
+          if (ts.isJsxOpeningElement(element) || ts.isJsxSelfClosingElement(element)) {
+            let astChanged = false;
+
+            const newProps = element.attributes.properties.map(prop => {
+              let newProp: JsxAttributeLike | undefined = prop;
+
+              if (ts.isJsxAttribute(prop)) {
+                const propName = prop.name.escapedText.toString();
+                const inputProp = widgetProperties.properties.find(l => l.name === propName);
+
+                if (inputProp) {
+                  if (prop.initializer) {
+                    if (ts.isStringLiteral(prop.initializer)) {
+                      if(inputProp.value !== prop.initializer.text) {
+                        newProp = factory.updateJsxAttribute(prop, prop.name, factory.createStringLiteral(inputProp.value));
+                        astChanged = true;
+                      }
+                    }
+                    else if (ts.isJsxExpression(prop.initializer)) {
+                      if (prop.initializer.expression) {
```

Comment:
could you do simple javascript funcion for each branch to make it intuitive? tooo many nested blocks

Reply:
created function for get attribute value:
- getAttributeValue
created functions for getting attribute value by type:
 - getStringTypeAttributeValue
 - getNumberTypeAttributeValue
 - getBooleanTypeAttributeValue